[CI] ExponentialHistogramBlockTests testComponentAccess failing

**Build Scans:**
- [elasticsearch-pull-request #105390 / part-3](https://gradle-enterprise.elastic.co/s/7qkcut3gwvsps)
- [elasticsearch-pull-request #105389 / part-3](https://gradle-enterprise.elastic.co/s/fuhgkygazpapa)
- [elasticsearch-intake #31486 / part3](https://gradle-enterprise.elastic.co/s/m5i5a6lszfxdw)
- [elasticsearch-intake #31485 / part3](https://gradle-enterprise.elastic.co/s/o3nibh2coopz4)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.data.ExponentialHistogramBlockTests.testComponentAccess" -Dtests.seed=8FFD6A2B2EAF0416 -Dtests.locale=sms -Dtests.timezone=America/Miquelon -Druntime.java=25
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.data.ExponentialHistogramBlockTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('testComponentAccess'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: <1>
     but: was <0>
```

**Issue Reasons:**
- [main] 3 consecutive failures in test testComponentAccess
- [main] 2 consecutive failures in step part-3
- [main] 2 consecutive failures in pipeline elasticsearch-pull-request
- [main] 4 failures in test testComponentAccess (2.2% fail rate in 179 executions)
- [main] 2 failures in step part-3 (1.8% fail rate in 109 executions)
- [main] 2 failures in step part3 (5.7% fail rate in 35 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (1.8% fail rate in 109 executions)
- [main] 2 failures in pipeline elasticsearch-intake (5.7% fail rate in 35 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -420,9 +420,6 @@ tests:
 - class: org.elasticsearch.xpack.security.authc.jwt.JwtRealmAuthenticateTests
   method: testJwkUpdatesByReloadWithFile
   issue: https://github.com/elastic/elasticsearch/issues/138397
-- class: org.elasticsearch.compute.data.ExponentialHistogramBlockTests
-  method: testComponentAccess
-  issue: https://github.com/elastic/elasticsearch/issues/138399
 
 # Examples:
 #

--- a/x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/mapper/ExponentialHistogramParserTests.java
+++ b/x-pack/plugin/analytics/src/test/java/org/elasticsearch/xpack/analytics/mapper/ExponentialHistogramParserTests.java
@@ -35,29 +35,31 @@ public class ExponentialHistogramParserTests extends ESTestCase {
     private static final String TEST_FIELD_NAME = "test_field";
 
     public void testParseRandomHistogram() throws IOException {
-        ExponentialHistogram histogram = ExponentialHistogramTestUtils.randomHistogram();
+        for (int i = 0; i < 20; i++) {
+            ExponentialHistogram histogram = ExponentialHistogramTestUtils.randomHistogram();
 
-        ExponentialHistogramParser.ParsedExponentialHistogram parsed;
-        try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
-            ExponentialHistogramXContent.serialize(builder, histogram);
-            String json = Strings.toString(builder);
-            parsed = doParse(json);
+            ExponentialHistogramParser.ParsedExponentialHistogram parsed;
+            try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+                ExponentialHistogramXContent.serialize(builder, histogram);
+                String json = Strings.toString(builder);
+                parsed = doParse(json);
+            }
+
+            List<IndexWithCount> expectedPositiveBuckets = IndexWithCount.fromIterator(histogram.positiveBuckets().iterator());
+            List<IndexWithCount> expectedNegativeBuckets = IndexWithCount.fromIterator(histogram.negativeBuckets().iterator());
+
+            assertThat(parsed.scale(), equalTo(histogram.scale()));
+
+            assertThat(parsed.zeroThreshold(), equalTo(histogram.zeroBucket().zeroThreshold()));
+            assertThat(parsed.zeroCount(), equalTo(histogram.zeroBucket().count()));
+
+            assertThat(parsed.positiveBuckets(), equalTo(expectedPositiveBuckets));
+            assertThat(parsed.negativeBuckets(), equalTo(expectedNegativeBuckets));
+
+            assertThat(parsed.sum(), equalTo(histogram.valueCount() == 0 ? null : histogram.sum()));
+            assertThat(parsed.min(), equalTo(Double.isNaN(histogram.min()) ? null : histogram.min()));
+            assertThat(parsed.max(), equalTo(Double.isNaN(histogram.max()) ? null : histogram.max()));
         }
-
-        List<IndexWithCount> expectedPositiveBuckets = IndexWithCount.fromIterator(histogram.positiveBuckets().iterator());
-        List<IndexWithCount> expectedNegativeBuckets = IndexWithCount.fromIterator(histogram.negativeBuckets().iterator());
-
-        assertThat(parsed.scale(), equalTo(histogram.scale()));
-
-        assertThat(parsed.zeroThreshold(), equalTo(histogram.zeroBucket().zeroThreshold()));
-        assertThat(parsed.zeroCount(), equalTo(histogram.zeroBucket().count()));
-
-        assertThat(parsed.positiveBuckets(), equalTo(expectedPositiveBuckets));
-        assertThat(parsed.negativeBuckets(), equalTo(expectedNegativeBuckets));
-
-        assertThat(parsed.sum(), equalTo(histogram.sum()));
-        assertThat(parsed.min(), equalTo(Double.isNaN(histogram.min()) ? null : histogram.min()));
-        assertThat(parsed.max(), equalTo(Double.isNaN(histogram.max()) ? null : histogram.max()));
     }
 
     public void testParseScaleMissing() {

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/ExponentialHistogramBlockTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/data/ExponentialHistogramBlockTests.java
@@ -95,9 +95,13 @@ public class ExponentialHistogramBlockTests extends ComputeTestCase {
                             }
                         }
                         case SUM -> {
-                            assertThat(componentBlock.getValueCount(i), equalTo(1));
-                            int valueIndex = componentBlock.getFirstValueIndex(i);
-                            assertThat(((DoubleBlock) componentBlock).getDouble(valueIndex), equalTo(histo.sum()));
+                            if (histo.valueCount() == 0) {
+                                assertThat(componentBlock.isNull(i), equalTo(true));
+                            } else {
+                                assertThat(componentBlock.getValueCount(i), equalTo(1));
+                                int valueIndex = componentBlock.getFirstValueIndex(i);
+                                assertThat(((DoubleBlock) componentBlock).getDouble(valueIndex), equalTo(histo.sum()));
+                            }
                         }
                         case COUNT -> {
                             assertThat(componentBlock.getValueCount(i), equalTo(1));

